tests/testthat/test-dbWriteTable.R

```diff
@@ -32,6 +32,14 @@ test_that("dbAppendTable() throws error if constraint violated", {
   expect_error(dbAppendTable(con, "t1", x), "added 7 rows|Duplicate entry")
 })
 
+test_that("dbAppendTable() works with Id", {
+  con <- mariadbDefault()
+  on.exit(dbDisconnect(con))
+
+  dbExecute(con, "create table T(n integer primary key)")
```

Comment:
Do we want a temporary table here?

Reply:
Just notice that the table will persist. Should I drop it in the end of the test case, or drop table if exists before creating it?

Reply:
I think drop table if exists makes good sense as it is similar with `dbWriteTable(overrite=TRUE)` in other test cases.

Reply:
`CREATE TEMPORARY TABLE` might be easier. Ideally all test cases should leave a clean state in the database.

Reply:
Yes, you are right.